implementation/src/main/java/io/smallrye/jwt/build/JwtEncryptionBuilder.java

```diff
@@ -25,7 +25,29 @@
      * @param algorithm the key encryption algorithm
      * @return JwtEncryptionBuilder
      */
-    JwtEncryptionBuilder keyEncryptionAlgorithm(KeyEncryptionAlgorithm algorithm);
+    JwtEncryptionBuilder keyAlgorithm(KeyEncryptionAlgorithm algorithm);
+
+    /**
+     * Set an 'alg' key encryption algorithm.
+     * Note that only 'RSA-OAEP-256' (default), 'ECDH-ES+A256KW' and 'A256KW' algorithms must be supported.
+     * A key of size 2048 bits or larger MUST be used with 'RSA-OAEP-256' algorithm.
+     *
+     * @param algorithm the key encryption algorithm
+     * @return JwtEncryptionBuilder
+     */
+    @Deprecated
```

Comment:
Can we add the tag to point to the method that should be used? It should be straightforward, but it would hurt :)